plugins/scaffolder-common/src/ScaffolderClient.ts

```diff
@@ -0,0 +1,418 @@
+            !this.scmIntegrationsApi.bitbucketServer.byHost(item.config.host),
+        ),
+      ...this.scmIntegrationsApi.bitbucketCloud.list(),
+      ...this.scmIntegrationsApi.bitbucketServer.list(),
+      ...this.scmIntegrationsApi.gerrit.list(),
+      ...this.scmIntegrationsApi.gitea.list(),
+      ...this.scmIntegrationsApi.github.list(),
+      ...this.scmIntegrationsApi.gitlab.list(),
+    ]
+      .map(c => ({ type: c.type, title: c.title, host: c.config.host }))
+      .filter(c => options.allowedHosts.includes(c.host));
+
+    return {
+      integrations,
+    };
+  }
+
+  /**
+   * {@inheritdoc ScaffolderApi.getTemplateParameterSchema}
+   */
+  async getTemplateParameterSchema(
+    templateRef: string,
+    options?: ScaffolderRequestOptions,
+  ): Promise<TemplateParameterSchema> {
+    return await this.requestRequired(
+      await this.apiClient.getTemplateParameterSchema(
+        {
+          path: parseEntityRef(templateRef, {
+            defaultKind: 'template',
+          }),
+        },
+        options,
+      ),
+    );
+  }
+
+  /**
+   * {@inheritdoc ScaffolderApi.scaffold}
+   */
+  async scaffold(
+    request: ScaffolderScaffoldOptions,
+    options?: ScaffolderRequestOptions,
+  ): Promise<ScaffolderScaffoldResponse> {
+    const response = await this.apiClient.scaffold(
+      {
+        body: request,
+      },
+      options,
+    );
+
+    if (response.status !== 201) {
+      const status = `${response.status} ${response.statusText}`;
+      const body = await response.text();
+      throw new Error(`Backend request failed, ${status} ${body.trim()}`);
+    }
+
+    const { id } = await response.json();
+    return { taskId: id };
+  }
+
+  /**
+   * {@inheritdoc ScaffolderApi.getTask}
+   */
+  async getTask(
+    taskId: string,
+    options?: ScaffolderRequestOptions,
+  ): Promise<ScaffolderTask> {
+    return await this.requestRequired(
+      await this.apiClient.getTask(
+        {
+          path: { taskId },
+        },
+        options,
+      ),
+    );
+  }
+
+  /**
+   * {@inheritdoc ScaffolderApi.streamLogs}
+   */
+  streamLogs(
+    request: ScaffolderStreamLogsOptions,
+    options?: ScaffolderRequestOptions,
+  ): Observable<LogEvent> {
+    if (this.useLongPollingLogs) {
+      return this.streamLogsPolling(request, options);
+    }
+
+    return this.streamLogsEventStream(request);
+  }
+
+  /**
+   * {@inheritdoc ScaffolderApi.dryRun}
+   */
+  async dryRun(
+    request: ScaffolderDryRunOptions,
+    options?: ScaffolderRequestOptions,
+  ): Promise<ScaffolderDryRunResponse> {
+    return await this.requestRequired(
+      await this.apiClient.dryRun(
+        {
+          body: {
+            template: request.template,
+            values: request.values,
+            secrets: request.secrets,
+            directoryContents: request.directoryContents,
+          },
+        },
+        options,
+      ),
+    );
+  }
+
+  private streamLogsEventStream({
```

Comment:
I couldn't use the generated API client for this one because it's not generic enough -- it uses `fetchEventSource` from `@microsoft/fetch-event-source`.